tools/env/fw_env.c

```diff
@@ -1097,10 +1099,33 @@ static int flash_io (int mode)
 	return rc;
 }
 
+int fw_env_open_real(struct env_opts *opts);
+
+int fw_env_open(struct env_opts *opts)
+{
+	int ret;
+	int nretries = 4;
+
+	while(nretries > 0) {
+		ret = fw_env_open_real(opts);
+		//Success!
+		if (ret == 0) {
+			return ret;
+		}
+		//Some error other than CRC
+		if (!crc_error_occurred) {
```

Comment:
This looks redundant, no harm in retrying for any error. You won't lose anything. Also put a sleep of 1 second between retries.

Reply:
Ok, I have made both changes.